rasa/core/training/interactive.py

```diff
@@ -793,11 +793,19 @@ def _write_stories_to_file(
     export_story_path: Text, events: List[Dict[Text, Any]], domain: Domain
 ) -> None:
     """Write the conversation of the conversation_id to the file paths."""
+    from rasa.shared.core.training_data.story_reader.yaml_story_reader import (
+        YAMLStoryReader,
+    )
 
     sub_conversations = _split_conversation_at_restarts(events)
 
     io_utils.create_path(export_story_path)
 
+    if YAMLStoryReader.is_yaml_story_file(export_story_path):
+        story_format = RASA_YAML
```

Comment:
`RASA_YAML` belongs to `shared.nlu.training_data.loading` package, i.e. it's NLU, not stories.
Instead we can create a writer here right away and pass it to the `tracker.export_stories`